chromium_src/components/search_engines/default_search_manager.cc

```diff
@@ -14,7 +15,12 @@ bool IsDefaultSearchProviderByExtension(PrefService* pref_service) {
 #if BUILDFLAG(IS_ANDROID)
   return false;
 #else
-  return pref_service->GetBoolean(prefs::kDefaultSearchProviderByExtension);
+  if (pref_service->HasPrefPath(prefs::kDefaultSearchProviderByExtension)) {
```

Comment:
@emerick I realized that `HasPrefPath()` is not appropriate.
Instead, we should use `pref_service->FindPreference(...)` here
`HasPrefPath()` returns false even it's regiserted path if its value is default.

Reply:
Makes sense, fixed!